### DISCUSSION TITLE
Container component with configurable background color.

### DISCUSSION THREAD

**Discussion by jsuarezgonz** [dxc-technology/halstack-react#1717] [Ideas] [answered]
One of the scenarios that is very likely to occur is the use of a container component such as our `DxcCard` or `DxcBox` with a configurable background color. 

For this scenario:

- Should we create a new component that allows this configuration?

- Should we give the user the option to configure the background color for any of these components, `DxcBox`, `DxcCard`, individually (apart from the theming)?

Some examples of other design systems:

[Seek ](https://seek-oss.github.io/braid-design-system/components/Box#backgrounds) The Box component has a `background` prop with a limited set of 'semantic' options.
[Lens](https://lens.loom.dev/components/container) The Container component has a `backgroundColor` prop limited to the use of one of the colors in their palette.
[Gestalt](https://gestalt.pinterest.systems/web/box) The Box component has a property `color` with limited values.

**Comment by GomezIvann** [comment]
This is a fascinating topic Jairo.

From my point of view, this use case should be fulfilled by the Box component, but not ours. I mean, we should consider reimplementing our Box as a more primitive component, with some of the most common properties of HTML/CSS properties of the containers, among which would be the background color. This component (as the Typography) should represent and be strongly related to our design tokens.

More examples:

- [Amplify](https://ui.docs.amplify.aws/react/components/view)
- [Atlassian](https://atlassian.design/components/primitives/box/examples)
- [Cloudscape](https://cloudscape.design/components/box/?tabId=playground)

**Reply by kukoc** [reply]
I was kind of convinced of refactoring the box as a primitive component, Exposing props controlling background and borders for example. That would help us to achieves thing like the chatbot in this screenshot: 

![image](https://github.com/dxc-technology/halstack-react/assets/7778817/ec0af532-1108-4323-ba99-4f6f48dae91e)


However, I don't like using again a box to show the top of the component. This is something that we would do in code with a Flex and using css for the color.

**Reply by GomezIvann** [reply]
From my point of view, that isn't a use case for a Box component. I think a well-implemented Card would solve it.

To illustrate with an example:
```
const ChatBot = () => (
  <DxcCard>
    <DxcCard.Header>
      <DxcBox backgroundColor="#5F249F" width="100%" height="100%">
        <DxcFlex justifyContent="end" gap="0.5rem">
          <DxcButton label="Volume" mode="text" />
          <DxcButton label="Expand" mode="text" />
        </DxcFlex>
      </DxcBox>
    </DxcCard.Header>
    <DxcCard.Body>
      {...}
    </DxcCard.Body>
    <DxcCard.Footer>
      <DxcFlex gap="1rem">
        <DxcTextInput />
        <DxcButton label="Send" />
        <DxcButton label="Mic" />
      </DxcFlex>
    </DxcCard.Footer>
  </DxcCard>
);
```

I could also have given a `display="flex"` to the box of the Card header but I thought this would be more semantic.

**Reply by kukoc** [reply]
Oh!! Agree. It's a better use case for a card. However, we are still missing some props to round the borders or to add a shadow. Aren't we?

**Reply by GomezIvann** [reply]
Yes. Also to propose an inner structure inside the card using compound components. The Card component needs to be reviewed and re-built. That's why I wrote "a well-implemented Card".

Anyways, the fact that they have more rounded corners than us doesn't mean we should offer a border-radius prop. I mean, that's a design decision and should rely on our side. We shouldn't always be adapting our components to their necessities if we do not consider them fair enough. Most of the cases should be the other way around.

**Reply by kukoc** [reply, marked as answer]
Final Proposal: 

- Box - Replace the Box with a new Container component as the lowest level of our Layout components with props at the primitive level to configure aspects like background or borders and deprecate the Box. 

- Card -  Redesign and reimplement this component to offer an opinionated structure to display a piece of information.
